virttest/migration_template.py

```diff
@@ -0,0 +1,563 @@
+                          'client_pwd': self.params['migrate_dest_pwd'],
+                          'server_ip': self.params['migrate_source_host'],
+                          'server_cn': self.params['migrate_source_host_cn'],
+                          'server_pwd': self.params['migrate_source_pwd']}
+
+        conn_class = {'tls': TLSConnection,
+                      'tcp': TCPConnection,
+                      'ssh': SSHConnection}
+
+        logging.debug("Setup libvirtd remote access env")
+        conn_obj = conn_class[self.params['remote_protocol']](conn_args)
+        conn_obj.conn_setup()
+        conn_obj.auto_recover = True
+        self.objs_list.append(conn_obj)
+        if self.params['migrate_vm_back'] == 'yes':
+            logging.debug("Setup libvirtd remote access env for reverse migration")
+            if self.params['remote_protocol'] == 'tls':
+                tls_args = {'ca_cakey_path': conn_obj.pki_CA_dir,
+                            'scp_new_cacert': 'no'}
+                back_conn_args.update(tls_args)
+            back_conn_obj = conn_class[self.params['remote_protocol']](back_conn_args)
+            back_conn_obj.conn_setup()
+            back_conn_obj.auto_recover = True
+            self.objs_list.append(back_conn_obj)
+
+        # Enable libvirtd remote port in firewalld
+        if self.remote_port:
+            logging.debug("Enable libvirtd remote port in firewalld on dst host")
+            remote_dict = {'server_ip': self.params['remote_ip'],
+                           'server_user': self.params['remote_user'],
+                           'server_pwd': self.params['remote_pwd']}
+            open_port_in_iptables(self.remote_port, server_dict=remote_dict,
+                                  session=self.remote_session)
+            if self.params['migrate_vm_back'] == 'yes':
+                logging.debug("Enable libvirtd remote port in firewalld on src host")
+                open_port_in_iptables(self.remote_port)
+
+    def _setup_qemu_tls(self):
+        """
+        Set up native encryption migration env
+        """
+        conn_args = {'server_ip': self.params['migrate_dest_host'],
+                     'server_cn': self.params['migrate_dest_host_cn'],
+                     'server_pwd': self.params['migrate_dest_pwd'],
+                     'client_ip': self.params['migrate_source_host'],
+                     'client_cn': self.params['migrate_source_host_cn'],
+                     'client_pwd': self.params['migrate_source_pwd']}
+
+        back_conn_args = {'client_ip': self.params['migrate_dest_host'],
+                          'client_cn': self.params['migrate_dest_host_cn'],
+                          'client_pwd': self.params['migrate_dest_pwd'],
+                          'server_ip': self.params['migrate_source_host'],
+                          'server_cn': self.params['migrate_source_host_cn'],
+                          'server_pwd': self.params['migrate_source_pwd']}
+
```

Comment:
Line 431 to 443 are duplicated with 381 to 393. So it is better to abstract them to upper layer as global variable or instant variable.

Reply:
I added a new method _setup_remote_connection_base() to wrap the common part of _setup_libvirtd_remote_access() and _setup_qemu_tls()